File: lista03W/zad1/Kwadrat.java
public class Kwadrat extends Czworokat {
  public Kwadrat(double bok) {
    super(bok, bok, bok, bok, 90);
  }
  
  public double getPole() {
    return bok1 * bok1;
  }
  
  public double getObwod() {
    return 4 * bok1;
  }
  
  public String getNazwa() {
    return "Kwadrat";
  }
}

File: lista03W/zad2/kwadrat.cpp
#include "kwadrat.hpp"

Kwadrat::Kwadrat(double bok) : Czworokat(bok, bok, bok, bok, 90) {
  if(bok <= 0)
    throw std::invalid_argument("Bok musi byc dodatni");
}

double Kwadrat::getPole() const {
  return bok1 * bok1;
}

double Kwadrat::getObwod() const {
  return 4 * bok1;
}

std::string Kwadrat::getNazwa() const {
  return "Kwadrat";
}